### PR TITLE
ENH: remove cookie banner from extra-eu countries with exceptions

### PR DESCRIPTION
**Pull request by eagerterrier**
concept

### CODE REVIEW

File: src/app/lib/utilities/cookieCountries/index.js
Comment (karinathomasbbc) [inline src/app/lib/utilities/cookieCountries/index.js:1]: Can this be a .ts file instead of .js?

File: src/server/index.jsx [hunk 1]
```diff
@@ -239,12 +239,17 @@ server.get(
...
 
       data.toggles = toggles;
       data.path = urlPath;
       data.timeOnServer = Date.now();
       data.showAdsBasedOnLocation = headers['bbc-adverts'] === 'true';
+      data.showCookieBannerBasedOnCountry = ['pidgin', 'hausa'].includes(
+        service,
+      )
+        ? showCookieBannerBasedOnCountry
+        : true;
```
Comment (karinathomasbbc) [inline src/server/index.jsx:252]: Is this just for testing / gradual rollout purposes?

File: src/server/index.jsx [hunk 2]
```diff
@@ -336,9 +341,11 @@ server.get(
           `https://www.bbcweb3hytmzhn5d532owbu6oqadra5z3ar726vq5kgwwn6aucdccrad.onion${urlPath}`,
         );
 
+        const allHeaders = ['X-country'];
```
Comment (karinathomasbbc) [inline src/server/index.jsx]: Could we rename this to `varyHeaders`?

File: src/app/legacy/containers/ConsentBanner/useConsentBanners/index.js [hunk 3]
```diff
@@ -125,21 +131,27 @@ const useConsentBanner = (isUK = false) => {
...
+      setUserDidDismissCookieBanner(isUK, 1);
+      setUserDidAcceptPolicy();
+      setUserDidSeePrivacyBanner(1);
     }
 
     if (!userHasPolicyCookie) {
       setDefaultPolicy();
     }
-  }, []);
+  }, [showCookieBannerBasedOnCountry]);
```
Comment (karinathomasbbc) [inline src/app/legacy/containers/ConsentBanner/useConsentBanners/index.js]: Can we fix this `useEffect` warning?

**Review by karinathomasbbc — CHANGES_REQUESTED** [review]

File: src/server/index.jsx (hunk 1 above)
Comment (eagerterrier) [reply src/server/index.jsx:252]: yes. We have hypotheses. Want to test them out. I'll share with you

File: src/server/index.jsx (hunk 2 above)
Comment (eagerterrier) [reply src/server/index.jsx]: oh yes - much better! Thanks

**Pull request merged**